handlers/transcode.go

```diff
@@ -72,9 +68,179 @@ func (d *CatalystAPIHandlersCollection) TranscodeSegment() httprouter.Handle {
 			return
 		}
 
-		if err := clients.DefaultCallbackClient.SendTranscodeStatusError(transcodeRequest.CallbackUrl, "NYI - not yet implemented"); err != nil {
-			errors.WriteHTTPInternalServerError(w, "error send transcode error", err)
+		if err := RunTranscodeProcess(d.MistClient, transcodeRequest); err != nil {
+			errors.WriteHTTPInternalServerError(w, "Error running Transcode process", err)
 		}
-		_, _ = io.WriteString(w, "OK") // TODO later
 	}
 }
+
+// stream from a source to a destination buffer while also printing
+func streamOutput(src io.Reader, dst *bytes.Buffer, out io.Writer) error {
+	mw := io.MultiWriter(dst, out)
+	s := bufio.NewReader(src)
+	for {
+		var line []byte
+		line, err := s.ReadSlice('\n')
+		if err == io.EOF && len(line) == 0 {
+			break
+		}
+		if err == io.EOF {
+			return fmt.Errorf("Improper termination: %v", line)
+		}
+		if err != nil {
+			return err
+		}
+		_, err = mw.Write(line)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// RunTranscodeProcess starts `MistLivepeeerProc` as a subprocess to transcode inputStream into renditionsStream.
+func RunTranscodeProcess(mistClient clients.MistAPIClient, request TranscodeSegmentRequest) error {
+	inputStream, renditionsStream := config.GenerateStreamNames()
+	if err := mistClient.AddStream(inputStream, request.SourceFile); err != nil {
+		return fmt.Errorf("error adding stream to Mist: %s", err)
+	}
+
+	configPayload, err := json.Marshal(configForSubprocess(request, inputStream, renditionsStream))
+	if err != nil {
+		return fmt.Errorf("ProcLivepeerConfig json encode: %s", err)
+	}
+	args := string(configPayload)
+
+	transcodeCommand := exec.Command(config.PathMistProcLivepeer, args, "--debug", "8", "--kickoff")
+
+	var stdout, stderr bytes.Buffer
```

Comment:
If buffers are unused, can we remove them? Maybe change `streamOutput()` to reflect this.

Reply:
Hmm, unclear on what you mean - can you clarify? The buffers are used to store stdout/stderr from `io.MultiWriter`

Reply:
Noticing we don't need to duplicate stream from `stdoutPipe`&`stderrPipe` into buffers. We just want it printed into log? Is this true? Maybe remnant of debugging?